src/ray/raylet/node_manager.cc

```diff
@@ -790,6 +790,13 @@ void NodeManager::TreatTaskAsFailed(const TaskSpecification &spec) {
 
 void NodeManager::SubmitTask(const Task &task, const Lineage &uncommitted_lineage,
                              bool forwarded) {
+  if (local_queues_.TaskInQueue(task.GetTaskSpecification().TaskId())) {
+    RAY_LOG(WARNING)
+        << "Submitted task " << task.GetTaskSpecification().TaskId()
+        << " is already queued. This is most likely due to spurious reconstruction.";
```

Comment:
minor: I would make it clear that this duplicate submitted task will not be executed.